index.d.ts

```diff
@@ -301,6 +302,12 @@ export class NgrokHttpTunnelBuilder {
   metadata(metadata: string): this
   /** Begin listening for new connections on this tunnel. */
   listen(): Promise<NgrokHttpTunnel>
+  /**
+   * Begin listening for new connections on this tunnel,
+   * and bind to a local socket so this tunnel can be
+   * passed directly into net.Server.listen.
+   */
+  bind(): Promise<NgrokHttpTunnel>
```

Comment:
Having both listen and bind feels weird to me... What do you get by calling listen and not bind? Do you lose the ability to forward connections if you call bind?

Maybe we should have `listen` *always* do the bind-and-forward routine that `bind` is doing now, and drop the forwarding methods from the Tunnel objects and instead hang them off of the builder. Then your options for building the tunnel are either:
1. Give me the listener and let me use it as such
2. Forward to this other thing

Hm, but then what do the forward methods return? My first thought was `Promise<void>` (and `Future<Output=()>` in the rust equivalent), but then you lose all of the tunnel information. Maybe you *just* get the tunnel information from the forward methods, and the actual forwarding is spun off into the background?

I think my takeaway here is that I need to actually sit down and play with that API and put together a real spec for tunnel forwarding 😛

Reply:
Agree it would be better to not have two ways to build tunnels. Maybe we move the behavior to an optional argument on `listen` as to whether to bind or not? I would at least want the user to be able to turn it off. I resisted silently binding when the user didn't explicitly ask for it, but maybe having it on by default isn't harmful. Looks like I can get away with moving the forwarding to be lazily done on the call to `handle` so other use-cases would still work.

I think moving `forward*` into the builders would return `Promise<*Tunnel>` and require the actual forwarding to be back-grounded on the rust side, as you say. I don't think there would be a downside to that but sometimes there's surprising behavior.

Reply:
Pushed a commit which has `listen` bind by default.